sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzMonList.cs

```diff
@@ -68,7 +68,7 @@ public static object GetTagValue(ref AzMonList list, string tagName)
 
             for (int i = 0; i < length; i++)
             {
-                if (list[i].Key == tagName)
+                if (ReferenceEquals(list[i].Key, tagName))
```

Comment:
Would this cause problems?

```csharp
using System;
using System.Text;
					
public class Program
{
	private const string Hello = "hello";

	public static void Main()
	{
		var a = Hello;
		var b = (new StringBuilder(Hello)).ToString();
		Console.WriteLine(object.ReferenceEquals(a, b));
	}
}
```

Reply:
`GetTagValue` takes [`SemanticConventions`](https://github.com/Azure/azure-sdk-for-net/blob/4162f6fa2445b2127468b9cfd080f01c9da88eba/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/SemanticConventions.cs) attributes as input key. This is pre-defined as const string in both `OpenTelemetry` and Azure Monitor Exporter. Hence this should not cause an issue.

But, we cannot deny the fact that if someone calls this `GetTagValue` like the sample above it might run into an issue. As this class is internal chances are very less for someone to call this incorrectly.

Reply:
@cijothomas need another pair of eyes here.